SPEC.md

```diff
@@ -0,0 +1,188 @@
+(since it is a soft-delete), so if you accidentally deleted the index, you could still restore it.
+
+
+### Vacuum
+The user can perform a hard-delete i.e., fully remove files and the metadata entry for a deleted index using the 
+vacuumIndex API. Once done, this action is irreversible as it physically deletes all the index files associated 
+with the index.
+
+
+### Restore
+A user can use the restoreIndex API to restore a deleted index. 
+This will bring back the latest version of index into ACTIVE status and makes it usable again for queries.
+
+
+### Cancel
+Cancel API to bring back index from an inconsistent state to the last known stable state. 
+E.g. if index fails during creation, in "CREATING" state. 
+The index will not allow any index modifying operations unless a cancel is called.
+
+> Note: Cancel from "VACUUMING" state will move it forward to "DOESNOTEXIST" state. 
+
+> Note: If no previous stable state exists, cancel will move it to "DOESNOTEXIST" state.
+
+# Index Type
+
+Hyperspace provides several index types.
+
+* Covering Index
+  * Roughly speaking, index data for [[CoveringIndex]] is just a vertical
+    slice of the source data, including only the indexed and included columns,
+    bucketed and sorted by the indexed columns for efficient access.
+* Data Skipping Index
+  * DataSkippingIndex is an index that can accelerate queries by filtering out
+    files in relations using sketches.
+
+# Querying with Hyperspace
+
+## Enable Hyperspace
+Hyperspace provides APIs to enable or disable index usage with Spark™.
+
+* By using enableHyperspace API, Hyperspace optimization rules become visible to the Apache Spark™ optimizer, 
+and it will exploit existing Hyperspace indexes to optimize user queries.
+* By using disableHyperspace command, Hyperspace rules no longer apply during query optimization. 
+You should note that disabling Hyperspace has no impact on created indexes as they remain intact.
+
+
+## List indexes
+You can use the indexes API which returns information about existing indexes as a Spark™'s DataFrame.
+For instance, you can invoke valid operations on this DataFrame for checking its content or analyzing it further 
+(for example, filtering specific indexes or grouping them according to some desired property).
+
+
+
+## Index Usage
+In order to make Spark™ use Hyperspace indexes during query processing, the user needs to make sure that Hyperspace 
+is enabled. After Hyperspace is enabled, without any change to your application code, Spark™ will use the indexes 
+automatically if it is applicable.
+
+
+## Explain
+Explains how indexes will be applied to the given dataframe.
+Explain API from Hyperspace is very similar to Spark's df.explain API but allows users to compare their original plan 
+vs the updated index-dependent plan before running their query. 
+You have an option to choose from html/plaintext/console mode to display the command output.
+
+
```

Comment:
nit: keep 2 new lines for consistency

Reply:
You mean blank lines? Ok I will review it. Thanks.

Reply:
I used 1 blnak line as default and used 2 lines only before headers.